### PR TITLE
Allow overriding of S3 region.

### PR DESCRIPTION
**Pull request by markdorison**
## Description
Allow overriding of S3 region.

## Motivation / Context
Database download would fail if bucket was not in default region.

## Testing Instructions / How This Has Been Tested
Tested locally.

### CODE REVIEW

File: src/Robo/Plugin/Commands/DevelopmentModeCommands.php [hunk 1]
```diff
@@ -163,26 +165,63 @@ protected function configureAwsCredentials(string $awsConfigDirPath, string $aws
...
+    /**
+     * Get S3 region for site.
+     *
+     * @param string $siteName
+     *   The site name.
+     *
+     * @return string
+     *   An S3 region.
+     */
+    protected function s3RegionForSite(string $siteName): string
```
Comment (adamzimmermann) [inline src/Robo/Plugin/Commands/DevelopmentModeCommands.php:213]: Should this leverage `s3BucketRequestConfig`?

File: src/Robo/Plugin/Commands/DevelopmentModeCommands.php [hunk 2]
```diff
@@ -163,26 +165,63 @@ protected function configureAwsCredentials(string $awsConfigDirPath, string $aws
...
         try {
-            $s3KeyPrefix = $this->getConfig('database_s3_key_prefix_string', $siteName);
-            $this->say("'$siteName' S3 Key prefix: '$s3KeyPrefix'");
-            $s3ConfigArray['Prefix'] = $s3KeyPrefix;
+            $region = $this->getConfig('database_s3_region', $siteName);
+            $this->say("'$siteName' database_s3_region set to $region.");
         } catch (TaskException $e) {
-            $this->say("No S3 Key prefix found for $siteName.");
+            // Set default region if one is not set.
+            $defaultRegion = 'us-east-1';
```
Comment (adamzimmermann) [inline src/Robo/Plugin/Commands/DevelopmentModeCommands.php]: Should this be a constant?

**Review by adamzimmermann — APPROVED** [review]

File: src/Robo/Plugin/Commands/DevelopmentModeCommands.php (hunk 1 above)
Comment (markdorison) [reply src/Robo/Plugin/Commands/DevelopmentModeCommands.php:213]: I don't follow.

Comment (adamzimmermann) [reply src/Robo/Plugin/Commands/DevelopmentModeCommands.php:213]: I was thinking that `s3BucketRequestConfig` was getting all config in a generic way, but I think that assumption was wrong. It's getting config but then building something unique.

**Review by mmatsoo — APPROVED** [review]
👍

File: src/Robo/Plugin/Commands/DevelopmentModeCommands.php (hunk 1 above)
Comment (markdorison) [reply src/Robo/Plugin/Commands/DevelopmentModeCommands.php:213]: s3BucketRequestConfig is also config needed for a request, whereas this is needed to instantiate the client.

**Pull request merged**